### DISCUSSION TITLE
Mosquitto + PVE 8.2.4:  broken?

### DISCUSSION THREAD

**Discussion by Zitt** [tteck/Proxmox#3439] [Help Needed] [answered]
I installed [MQTT ( Mosquito ) ](https://tteck.github.io/Proxmox/#mqtt-lxc) using all defaults and I'm getting some weird unexpected behavior. When launching the VM;  I get logged into the VM as root; but, nothing seems to be working quiet right.

Example: _root@mqtt:~# dmesg_
returns:
dmesg: read kernel buffer failed: Operation not permitted

Things appear to be somewhat working at the commandline; but mosquito_sub and mosquito_pub commands return:
Error: No such file or directory

given dmesg isn't working in the image; I haven't been able to do any debug.

thoughts?

**Comment by tteck** [comment]
Did you follow the post install instructions?

**Reply by Zitt** [reply]
yes. I created a user; which I assumed was for mqtt topics / etc. Not for logging into the VM.

**Reply by tteck** [reply]
It's a container, not a virtual machine.

**Reply by tteck** [reply, marked as answer]
Check the [Post Install](https://github.com/tteck/Proxmox/discussions/2921) again, I updated the instructions.

**Reply by tteck** [reply]
The command below will show what ports are listening (look for `*:1883 (LISTEN)`)
```
lsof -i -P -n | grep LISTEN
```

**Reply by tteck** [reply]
BTW, `dmesg` will not work in an unprivileged container.

**Reply by Zitt** [reply]
> BTW, `dmesg` will not work in an unprivileged container.

Oh? I just learned something. thanks!

> ```
> lsof -i -P -n | grep LISTEN
> ```

mosquitto 272       mosquitto    5u  IPv4 4446939      0t0  TCP *:1883 (LISTEN)
so that's should be good. 

Trying to figure out why I'm getting:
```
mosquitto_sub -d -t 'topic' -u picofan -p *****
```
Error: No such file or directory

I think it pure user error so I'm going to have to look for a baby step guide somewhere.
